src/StreamJsonRpc/JsonRpc.cs

```diff
@@ -946,11 +960,34 @@ private JsonRpcMessage HandleInvocationTaskResult(JToken id, Task t)
 
             if (t.IsFaulted)
             {
+                if (t.Exception != null)
```

Comment:
This is something you can assume without checking: `Task.IsFaulted === (Task.Exception != null)` is a guarantee.